### PR TITLE
Refactor sample app to deploy and run against real AWS

### PR DESCRIPTION
**Pull request by whummer**
Refactor sample app to deploy and run against real AWS.

Summary of changes:
* Adjust IAM policies to grant access to resources
* Adjust ARNs to match real AWS
* Allow specifying a custom API Gateway endpoint in the UI

Screenshot from running against real AWS:
image

### CODE REVIEW

**Comment by whummer** [comment]
Work in progress.. Still need to look into the failing test

File: demo/lambdas/worker.rb [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
+            sm_arn = sm.state_machine_arn
+        end
+    end
+
     records = event['Records']
-    client = Aws::States::Client.new(:endpoint => $endpoint)
     for rec in records do
         result = client.start_execution({
-            state_machine_arn: $state_machine_arn,
+            state_machine_arn: sm_arn,
```
Comment (baermat) [inline demo/lambdas/worker.rb:29]: Will this "blow up" if the 'processingStateMachine' is for whatever reason not present (= no arn will be set)?

Comment (whummer) [reply demo/lambdas/worker.rb:29]: Yep, good point - it definitely would be blowing up. We could also inject the SFn ARN into the Lambda environment - let me make that change quickly.. 👍

**Comment by whummer** [comment]
FYI - Should be fixed, going to merge @baermat 👍

**Pull request merged**